File: 실버이하의 문제들 모음/실버 100문제 풀기/브루투 포스/15649 n과 m(1)/15649.js
const fs = require('fs');
const input = fs.readFileSync('15649.txt').toString().split(' ')

const N = Number(input[0]);
const M = Number(input[1]);

const visited = new Array(N+1);

for (let i=0; i<=N; i++){
    visited[i] = false
}

let output = [];

function print() {
  console.log(output.join(' '));
}

function dfs(cnt) {
  if (cnt === M) {
    print();
    return;
  }

  for (let i = 1; i <= N; i++) {
    if (visited[i] === false){
        visited[i] = true;
        output.push(i);
        cnt += 1
        dfs(cnt);
        cnt -= 1
        output.pop();
        visited[i] = false;
        }
    }
}

dfs(0);

File: 실버이하의 문제들 모음/실버 100문제 풀기/브루투 포스/15649 n과 m(1)/15649.py
import sys
sys.stdin = open('15649.txt','r')

def dfs(now):
    global n,m,visit,arr

    if now == m:
        for a in arr:
            print(a, end = ' ')
        print()
        return


    for nxt in range(1,n+1):
        if visit[nxt] == 0:
            visit[nxt] = 1
            now += 1
            arr.append(nxt)
            dfs(now)
            arr.pop()
            now -= 1
            visit[nxt] = 0

    return


n,m = map(int, input().split())
visit = [0]*(n+1)
arr = []

dfs(0)